src/main/java/org/apache/pirk/querier/wideskies/encrypt/EncryptQuery.java

```diff
@@ -178,11 +217,14 @@ public Querier encrypt(int numThreads) throws InterruptedException, PIRException
       {
         // Hash collision
         selectorQueryVecMapping.clear();
-        hashKey = queryInfo.getHashKey() + ++keyCounter;
+        hashKey = hashKeyBase + getRandByteString(10);
         logger.debug("index = " + index + "selector = " + selector + " hash collision = " + hash + " new key = " + hashKey);
         index = -1;
       }
     }
+    
+    // Save off the final hashKey that we ended up using
+    queryInfo.setHashKey(hashKey);
```

Comment:
I don't think there's any plausible situation in which the `QueryInfo` object is used more than once in a single run of Pirk. Therefore, editing the `QueryInfo` shouldn't be a problem. More than one operation on the same parameters would be performed using multiple runs of the JAR with the same backing properties files.